### ISSUE TITLE
`object-property-newline` does not work for object destructuring. 

### ISSUE THREAD

**Issue by crazyjat** [eslint-stylistic/eslint-stylistic#194]
### Describe the bug

I have my rule setup as follows:

```
        "@stylistic/object-property-newline": [
            "error",
            {
                "allowAllPropertiesOnSameLine": false
            }
        ],
```

but my code does not show an error:

```
    const {
        foo, bar, foobar,
    } = someOtherVariable;
```

### Reproduction

Use example provided

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by antfu** [comment]
I am not sure if we eventually would be able to fix it, but before that you could try this plugin instead: https://github.com/antfu/eslint-plugin-antfu/blob/main/src/rules/consistent-list-newline.md

**Comment by antfu** [comment]
Let's discuss at https://github.com/eslint-stylistic/eslint-stylistic/issues/173.
